test/semester.test.js

```diff
@@ -94,3 +101,54 @@ test('dateInSemester()', t => {
   t.equal(s.dateInSemester(semester, dateOut), false, 'returns false for date outside of the semester')
   t.end()
 })
+
+test('getPeriodFromDate()', t => {
+  const periods = [
+    {type: 'teaching', starts_at: '2015-01-01', ends_at: '2015-02-01', first_week_parity: 'even'},
+    {type: 'exam',     starts_at: '2015-03-01', ends_at: '2015-04-01', first_week_parity: 'even'},
+    {type: 'teaching', starts_at: '2015-04-01', ends_at: '2015-06-01', first_week_parity: 'even'},
+    {type: 'teaching', starts_at: '2015-06-01', ends_at: '2017-01-01', first_week_parity: 'even'},
+  ]
+
+  t.deepEqual(s.getPeriodFromDate(new Date('2015-01-01'), periods), periods[0], 'is within period')
+  t.deepEqual(s.getPeriodFromDate(new Date('2015-01-15'), periods), periods[0], 'is within period')
+  t.deepEqual(s.getPeriodFromDate(new Date('2015-03-15'), periods), periods[1], 'is within period')
+  t.deepEqual(s.getPeriodFromDate(new Date('2016-01-01'), periods), periods[3], 'is within period')
+
+  t.deepEqual(s.getPeriodFromDate(new Date('1970-05-15'), periods), null, 'is not within any period')
+  t.deepEqual(s.getPeriodFromDate(new Date('2015-02-15'), periods), null, 'is not within any period')
+  t.deepEqual(s.getPeriodFromDate(new Date('2020-05-15'), periods), null, 'is not within any period')
```

Comment:
Duuude… Can you put this into a cycle and parametrize the test? This is terrible to look at.